boat_simulator/nodes/physics_engine/fluid_forces.py

```diff
@@ -72,9 +166,141 @@ def interpolate(self, attack_angle: Scalar) -> Tuple[Scalar, Scalar, Scalar]:
                     area is expressed in square meters (m^2).
         """
 
-        # TODO: Implement this method using `np.interp`.
+        lift_coefficient = np.interp(
+            attack_angle, self.__lift_coefficients[:, 0], self.__lift_coefficients[:, 1]
+        )
+        drag_coefficient = np.interp(
+            attack_angle, self.__drag_coefficients[:, 0], self.__drag_coefficients[:, 1]
+        )
+        area = np.interp(attack_angle, self.__areas[:, 0], self.__areas[:, 1])
+        return lift_coefficient, drag_coefficient, area
+
+    def draw_boat(ax, position, orientation):
+        """Draws a simplified boat shape on the given axes, ensuring it
+        aligns with the orientation line."""
+        boat_length = 1.0
+        boat_width = 0.3
+
+        # Center the shape around (0, 0)
+        front_extension = 3 * boat_length / 4
+        rear_extension = -boat_length / 4
+
+        # Calculate the offset to center the shape
+        offset = front_extension + rear_extension
+
+        boat_shape = np.array(
+            [
+                [front_extension - offset, 0],
+                [boat_length / 2 - offset, boat_width / 2],
+                [rear_extension - offset, 0],
+                [boat_length / 2 - offset, -boat_width / 2],
+                [front_extension - offset, 0],
+            ]
+        )
+
+        # Rotation matrix for anticlockwise rotation
+        rotation_matrix = np.array(
+            [
+                [np.cos(np.deg2rad(orientation)), np.sin(np.deg2rad(orientation))],
+                [np.sin(np.deg2rad(orientation)), np.cos(np.deg2rad(orientation))],
+            ]
+        )
+
+        # Apply rotation
+        rotated_boat = np.dot(boat_shape, rotation_matrix)
+
+        # Translate boat to its position
+        translated_boat = rotated_boat + np.array(position)
+
+        # Draw the boat
+        ax.plot(translated_boat[:, 0], translated_boat[:, 1], "k")
+
+        # Ensure the orientation line is drawn correctly
+        # Calculate a point along the orientation direction
+        direction = np.array([np.cos(np.deg2rad(orientation)), np.sin(np.deg2rad(orientation))])
+        line_start = np.array(position)
+        line_end = (
+            line_start + direction * boat_length
+        )  # Extend the line out from the boat's position
+
+        # Draw orientation line
+        ax.plot(
+            [line_start[0], line_end[0]], [line_start[1], line_end[1]], "black", linestyle="--"
+        )
+
+    def visualize_forces(
+        self, apparent_velocity, lift_force, drag_force, position=[0, 0], orientation=0
+    ):
```

Comment:
Does this function belong in this class?

Reply:
Where should it go?